### PR TITLE
Feature selenium grid

### PR DESCRIPTION
**Pull request by kthakore**

### CODE REVIEW

File: mvp/selenium_mvp.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+const logging = require('selenium-webdriver/lib/logging')
+const sleep = require('await-sleep');
+
+const cheerio = require('cheerio')
+const _ = require('lodash')
+const async = require('async')
+
+const fs = require('fs')
+
+const processIframes = async (driver, ) => {
```
Comment (even-steven) [inline mvp/selenium_mvp.js]: Is this used?

I would have thought that this syntax is invalid

File: mvp/selenium_mvp.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+          }
+        // let loading animation finish
+        
+        const microBiomeScore = await driver.findElement(By.css('span.score')).getText()
+        const percentile = await driver.findElement(By.xpath(`//section[contains(@class, 'microbiome-diversity__percentile')]/aside[contains(@class, 'number')]`)).getText()
+        
+        _.merge(result, {"microdiversityScore": microBiomeScore, "percentile": percentile})
+
+        } catch (e) {
+          //console.log("COULDN'T find score in right iframe", e)
```
Comment (even-steven) [inline mvp/selenium_mvp.js]: Should we just log the error here, since we're already catching it

File: mvp/selenium_mvp.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    await sleep(5000)
+    _.each(iframes, async (iframe) => {    // Go through each u_app iframe and get ids for it
+      
+        try {
+
+          await driver.switchTo().frame(iframe);
+
+          await sleep(5000)
+          try { 
+            await driver.wait(until.elementLocated(By.css('span.score')), 20000)
```
Comment (even-steven) [inline mvp/selenium_mvp.js]: If this fails, then the other  (presumably). Might as well remove the try, and let the encapsulating try-catch, catch the error.

e.g:

```
try {
    await driver.switchTo().frame(iframe);
    await sleep(5000)
    await driver.wait(until.elementLocated(By.css('span.score')), 20000)
    const microBiomeScore = await driver.findElement(By.css('span.score')).getText()
    const percentile = await driver.findElement(By.xpath(`//section[contains(@class, 'microbiome-diversity__percentile')]/aside[contains(@class, 'number')]`)).getText()
    _.merge(result, {"microdiversityScore": microBiomeScore, "percentile": percentile})
} catch (e) {
      console.log('Error collecting microbiome diversity percentile', e);
}
```

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ai.doc.neuron.app.ubiome",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (TheSuperFly) [inline package.json]: Good thing you thought about that. :D

File: mvp/selenium_mvp.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+          const wsData = _.get(message, 'params.response.payloadData')
+          
+         // const { resultWS } = parseSocket(wsData);
+          const targets = new RegExp(/diversity|probiotics|comps20/)
+          if (targets.test(wsData)) {
+            const match = wsData.match(targets)
+            if (match[0] !== undefined) {
+              const websocketData = _.get(result, ['websocketData'], [])
+              try {
```
Comment (even-steven) [inline mvp/selenium_mvp.js]: this try-catch isn't needed, I think

File: mvp/selenium_mvp.js (hunk 3 above)
Comment (kthakore) [reply mvp/selenium_mvp.js]: No no I want the logs no?

Comment (even-steven) [reply mvp/selenium_mvp.js]: the log can be performed in the encapsulating catch where you had:

```catch (e) {
          //console.log("COULDN'T find score in right iframe", e)```

File: src/index.js [hunk 6]
```diff
@@ -8,9 +9,13 @@ async function main() {
         name: 'uBiome Explorer',
         groups: 'microbiome',
         apiBaseUrl: 'https://apps.ubiome.com',
-        wsUrl: 'wss://explorer.ubiome.com/sockjs/572/6wejcx7t/websocket'
-      }
-    }
+        wsUrl: 'wss://explorer.ubiome.com/sockjs/572/6wejcx7t/websocket',
+        selenium: {
+          host: 'http://localhost:4444/wd/hub',
```
Comment (nkashy1) [inline src/index.js]: The HOST and PORT should come from environment variables that should be defined in the kubernetes manifest. Is there some sort of magic which will overwrite `selenium.host` here?

File: src/importers/strategies/Selenium.js [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+ *  to get data from the uBiome meteor App
+ */
+export default class Selenium {
+  /**
+   *
+   * @param {Any} config
+   */
+  constructor(config: Any, log: Any) {
+    this.SELENIUM_SERVER_HOST = _.get(
+      config,
```
Comment (nkashy1) [inline src/importers/strategies/Selenium.js]: @kthakore : What kind of magic is going on here? What does the `config` object contain? Can you please improve the jsdoc string on top to reflect the structure of `config` or to at least elucidate the behaviour of this `_.get`.

Comment (nkashy1) [reply src/importers/strategies/Selenium.js]: I think this invalidates my earlier comment.

File: src/index.js (hunk 6 above)
Comment (kthakore) [reply src/index.js]: Yep that RC thing does.

File: src/importers/strategies/Selenium.js (hunk 7 above)
Comment (kthakore) [reply src/importers/strategies/Selenium.js]: Yep so basically the config object is "magic" cause of how RC works. Node-neuron-app injects the config into the Importer object which I then pass to this class. The config object is "some" what defined in src/index.js (see custom config). The rest of it is defined in `node-neuron-app`. It is really really really hard to grok and I hate it :(

File: mvp/selenium_mvp.js (hunk 5 above)
Comment (kthakore) [reply mvp/selenium_mvp.js]: It is as it makes the failure there not make the next selenium await fail.

File: mvp/selenium_mvp.js (hunk 3 above)
Comment (kthakore) [reply mvp/selenium_mvp.js]: let me look at why I had this. All I know is I needed the try catches

**Pull request merged**